content/en/developers/dogstatsd/unix_socket.md

```diff
@@ -16,161 +16,222 @@ further_reading:
 
-To enable the Agent DogStatsD server:
+{{< tabs >}}
+{{% tab "Host" %}}
 
-1. Edit the [Agent's main configuration file][2] to set `dogstatsd_socket` to the path where DogStatsD should create its listening socket:
+1. Edit the [Agent's main configuration file][1] to set `dogstatsd_socket` to the path where DogStatsD should create its listening socket:
 
     ```yaml
     ## @param dogstatsd_socket - string - optional - default: ""
-    ## Listen for Dogstatsd metrics on a Unix Socket (*nix only). Set to a valid filesystem path to enable.
+    ## Listen for Dogstatsd metrics on a Unix Socket (*nix only).
+    ## Set to a valid filesystem path to enable.
     #
     dogstatsd_socket: '/var/run/datadog/dsd.socket'
     ```
 
-2. [Restart your Agent][3].
+2. [Restart your Agent][2].
 
-**Note**: You can also set the socket path with the `DD_DOGSTATSD_SOCKET` environment variable for the container Agent.
 
-### DogStatsD Client
+[1]: /agent/guide/agent-configuration-files/#agent-main-configuration-file
+[2]: /agent/guide/agent-commands
+{{% /tab %}}
+{{% tab "Docker" %}}
 
-#### Native support in client libraries
+1. Set the socket path with the `DD_DOGSTATSD_SOCKET=<YOUR_UDS_PATH>` environment variable on the Agent container.
 
-The following official DogStatsD client libraries natively support UDS traffic. Refer to the library's documentation on how to enable UDS traffic. **Note**: As with UDP, enabling client-side buffering is highly recommended to improve performance on heavy traffic:
+2. The socket file then needs to be accessible to the client containers. To achieve this, Datadog recommends mounting a host directory on both sides (read-only in your client containers, read-write in the Agent container). Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts:
 
-| Language | Library                              |
-| -------- | ------------------------------------ |
-| Golang   | [DataDog/datadog-go][4]              |
-| Java     | [DataDog/java-dogstatsd-client][5]   |
-| Python   | [DataDog/datadogpy][6]               |
-| Ruby     | [DataDog/dogstatsd-ruby][7]          |
-| PHP      | [DataDog/php-datadogstatsd][8]       |
-| C#       | [DataDog/dogstatsd-csharp-client][9] |
+    - Start the Agent container with `-v /var/run/datadog:/var/run/datadog`
+    - Start your application containers with `-v /var/run/datadog:/var/run/datadog:ro`
+
+{{% /tab %}}
+{{% tab "Kubernetes" %}}
 
-#### Using netcat
+1. Set the socket path with the `DD_DOGSTATSD_SOCKET=<YOUR_UDS_PATH>` environment variable on the Agent container.
+
+2. The socket file then needs to be accessible to the client containers. To achieve this, Datadog recommends mounting a host directory on both sides (read-only in your client containers, read-write in the Agent container). Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts.
```

Comment:
```suggestion
2. Make the socket file accessible to the client containers by mounting a host directory on both sides (read-only in your client containers and read-write in the Agent container). Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts.
```